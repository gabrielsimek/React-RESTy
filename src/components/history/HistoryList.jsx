import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';

const HistoryList = ({ history }) => {
  const historyItems = history.map(item => {
    return <HistoryItem 
      key={item.id}
      {...item}
    />;
  });
  return <ul>{historyItems}</ul>;
};

HistoryList.propTypes = {
  history: PropTypes.array.isRequired
};

export default HistoryList;
