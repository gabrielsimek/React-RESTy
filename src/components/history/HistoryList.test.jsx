import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList';

describe('HistoryList component', () => {
  afterEach(() => cleanup());
  const onClick = jest.fn();
  it('renders HistoryList', () => {
    const { asFragment } = render(<HistoryList onClick={onClick}  history={ [{ body: null, id: 0, method: 'GET', url: 'https://hey-arnold-api.herokuapp.com/api/v1/characters?perPage=3&page=2' }]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
