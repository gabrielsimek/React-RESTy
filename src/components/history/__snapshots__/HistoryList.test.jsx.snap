// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders HistoryList 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-direction: column; align-items: center;"
  >
    <h2>
      History
    </h2>
    <ul
      style="margin: 10px; list-style: none; width: 400px;"
    >
      <li
        style="cursor: pointer; border: 1px solid black; margin: 5px; padding: 5px;"
      >
        <h2>
          GET
        </h2>
        <p>
          https://hey-arnold-api.herokuapp.com/api/v1/characters?perPage=3&page=2
        </p>
        <p />
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
